### PR TITLE
feat(ec2): Ensure not default Network Access Control Lists are used

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context
Unused network ACLs that are not associated with any subnets can lead to unnecessary complexity and potential security risks. Ensuring that only actively used network ACLs are present helps maintain a cleaner and more secure network configuration.

In order to ensure that I added a new check that will fail if a network ACL is not associated with any subnet, except for default network ACLs which are exempt.

### Description
Added new `ec2_networkacl_unused_not_default` check with respective unit tests. Modify `ec2_service` to change `network_acls` from list to dictionary and adjust 3 old checks `ec2_networkacl_allow_ingress_any_port, ec2_networkacl_allow_ingress_tcp_port_22, ec2_networkacl_allow_ingress_tcp_port_3389 `. Added two attributes `default` and `in_use` to `NetworkAcl` model and change the service test to adjust to that changes.

### Checklist

- Are there new checks included in this PR? Yes 
    - If so, do we need to update permissions for the provider? No
- [x] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/ec2/ec2_networkacl_unused_not_default/ec2_networkacl_unused_not_default.py
Comment (MrCloudSec) [inline prowler/providers/aws/services/ec2/ec2_networkacl_unused_not_default/ec2_networkacl_unused_not_default.py:1]: Why `network_acl.name if network_acl.name else network_acl.id` if there is always an ID?

File: prowler/providers/aws/services/ec2/ec2_service.py [hunk 1]
```diff
@@ -156,15 +156,20 @@ def _describe_network_acls(self, regional_client):
...
-                                id=nacl["NetworkAclId"],
-                                arn=arn,
-                                name=nacl_name,
-                                region=regional_client.region,
-                                entries=nacl["Entries"],
-                                tags=nacl.get("Tags"),
-                            )
+                        in_use = False
+                        for subnet in nacl["Associations"]:
+                            if subnet["SubnetId"] != "None":
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ec2/ec2_service.py]: Why "None" ?

File: prowler/providers/aws/services/ec2/ec2_networkacl_unused_not_default/ec2_networkacl_unused_not_default.metadata.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "ec2_networkacl_unused_not_default",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ec2/ec2_networkacl_unused_not_default/ec2_networkacl_unused_not_default.metadata.json]:
```suggestion
  "CheckID": "ec2_networkacl_unused",
```

File: prowler/providers/aws/services/ec2/ec2_networkacl_unused_not_default/ec2_networkacl_unused_not_default.py
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/ec2/ec2_networkacl_unused_not_default/ec2_networkacl_unused_not_default.py:1]: All the others Network ACL checks were using that structure. I felt it a bit weird at first but it makes sense if you want to show the name of the Network ACL although sometimes it has no name and you can only show the id.

File: prowler/providers/aws/services/ec2/ec2_service.py (hunk 1 above)
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/ec2/ec2_service.py]: First thing that came to my mind. Looking at it now maybe is better to do `if subnet["SubnetId"]:`.

**Comment by MarioRgzLpz** [comment]
@sergargar Rename done in commit [fe9da3b](https://github.com/prowler-cloud/prowler/pull/4917/commits/fe9da3be4b5a919f4df3d1245404c575f5fae973) and change if condition in commit [5c5b12f](https://github.com/prowler-cloud/prowler/pull/4917/commits/5c5b12f1b43b7b5f5ee7b652be889edaabb82701)

File: prowler/providers/aws/services/ec2/ec2_networkacl_unused/ec2_networkacl_unused.metadata.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
+  "Severity": "low",
+  "ResourceType": "AwsEc2NetworkAcl",
+  "Description": "Ensure that there are no unused network access control lists (network ACLs) in your virtual private cloud (VPC). The control fails if the network ACL isn't associated with a subnet. The control doesn't generate findings for an unused default network ACL.",
+  "Risk": "Unused network ACLs may represent a potential security risk if left in place without purpose, as they could be mistakenly associated with subnets later.",
+  "RelatedUrl": "https://docs.aws.amazon.com/config/latest/developerguide/vpc-network-acl-unused-check.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ec2/ec2_networkacl_unused/ec2_networkacl_unused.metadata.json]:
```suggestion
      "CLI": "aws ec2 delete-network-acl --network-acl-id <nacl_id>",
```

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**